src/main.rs

```diff
@@ -121,7 +121,14 @@ fn main() {
     let socket = bind_socket(local_addr);
     let (sink, stream) = UdpFramed::new(socket, DhtCodec).split();
 
-    let (dht_pk, dht_sk) = load_or_gen_keys(cli_config.keys_file);
+    let (dht_pk, dht_sk) = if let Some(sk) = cli_config.sk {
+        (sk.public_key(), sk)
+    } else if let Some(keys_file) = cli_config.keys_file {
+        load_or_gen_keys(keys_file)
+    } else {
+        warn!("Neither secret key nor keys file is specified, using temporary generated keys pair");
```

Comment:
Well, better add a flag `--temp-key`.

Reply:
Then maybe just require one of `secret-key` or `key-file`?